### PR TITLE
Integration test for InstanceAdmin::ListInstances.

### LINKED ISSUE #343: Create integration test for InstanceAdmin::ListInstances
While it is not possible to test against the emulator we can test against the production server.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #343.

### CODE REVIEW

File: bigtable/tests/instance_admin_integration_test.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  }
+}
+
+int main(int argc, char* argv[]) {
+  ::testing::InitGoogleTest(&argc, argv);
+
+  if (argc != 2) {
+    std::string const cmd = argv[0];
+    auto last_slash = std::string(cmd).find_last_of('/');
+    // Show Usage if invalid no of arguments
```
Comment (mbrukman) [inline bigtable/tests/instance_admin_integration_test.cc]: s/no/number/

File: bigtable/tests/instance_admin_integration_test.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+ private:
+  static std::string project_id_;
+};
+
+std::string InstanceTestEnvironment::project_id_;
+
+class InstanceAdminIntegrationTest : public testing::Test {
+ protected:
+  std::unique_ptr<bigtable::InstanceAdmin> instance_admin_;
```
Comment (mbrukman) [inline bigtable/tests/instance_admin_integration_test.cc]: Put member variables last, after methods?

File: bigtable/tests/instance_admin_integration_test.cc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  static std::string const& project_id() { return project_id_; }
+
+ private:
+  static std::string project_id_;
+};
+
+std::string InstanceTestEnvironment::project_id_;
+
+class InstanceAdminIntegrationTest : public testing::Test {
```
Comment (mbrukman) [inline bigtable/tests/instance_admin_integration_test.cc]: `s/testing::Test/::testing::Test/` ?

Above, `::testing::Environment` uses absolute namespace reference.

File: bigtable/tests/instance_admin_integration_test.cc (hunk 2 above)
Comment (coryan) [reply bigtable/tests/instance_admin_integration_test.cc]: Done.

File: bigtable/tests/instance_admin_integration_test.cc (hunk 1 above)
Comment (coryan) [reply bigtable/tests/instance_admin_integration_test.cc]: Done.

File: bigtable/tests/instance_admin_integration_test.cc (hunk 3 above)
Comment (coryan) [reply bigtable/tests/instance_admin_integration_test.cc]: Done.

**Pull request merged**